provider/src/test/kotlin/org/gradle/kotlin/dsl/integration/GradleKotlinDslIntegrationTest.kt

```diff
@@ -602,11 +602,44 @@ class GradleKotlinDslIntegrationTest : AbstractIntegrationTest() {
                 Build file '${buildFile.canonicalPath}' line: 1
 
                 * What went wrong:
-                Unresolved reference: foo
+                Script compilation error:
+
+                  1: foo
+                     ^- Unresolved reference: foo
+
+                1 error
             """.replaceIndent())
         )
     }
 
+    @Test
+    fun `multiple script compilation errors message`() {
+        val buildFile = withBuildScript("println(foo)\n\n\n\n\nprintln(\"foo\").bar.bazar\n\n\n\nprintln(cathedral)")
+
+        assertThat(
+            buildFailureOutput().convertLineSeparators(),
+            containsString("""
+                FAILURE: Build failed with an exception.
+
+                * Where:
+                Build file '${buildFile.canonicalPath}' line: 1
+
+                * What went wrong:
+                Script compilation errors:
+
+                  01: println(foo)
+                              ^- Unresolved reference: foo
```

Comment:
👍 Very nice!

Reply:
This is absolutely awesome! Wondering if it's helpful to have it say `Line 01: println(foo)` instead of just `01: println(foo)`

Reply:
@JLLeitschuh, I was wondering the same. Something is missing to explain what these numbers are.

```
Line 01: println(foo)
                 ^- Unresolved reference: foo
```

or, shorter:
```
L01: println(foo)
             ^- Unresolved reference: foo
```

Reply:
Either one is better than not having the information at all. It's really confusing just seeing numbers without the label. For all the user knows they could be error codes.

Reading the test was initially confusing until I figured out that they were line numbers. You want the information to be as immediately apparent as possible.

Reply:
I opted for the verbose and explicit option.
```
Line 01: println(foo)
                 ^- Unresolved reference: foo
```

Reply:
👍  I think this was the right call.